components/helm-broker/internal/controller/addonsconfiguration_controller.go

```diff
@@ -0,0 +1,516 @@
+			repo.FetchingError(err)
+			repositories.AddRepository(repo)
+
+			r.log.Errorf("url %q address is not valid: %s", specRepository.URL, err)
+			continue
+		}
+
+		adds, err := r.createAddons(specRepository.URL)
+		if err != nil {
+			repo.FetchingError(err)
+			repositories.AddRepository(repo)
+
+			r.log.Errorf("while creating addons for repository from %q: %s", specRepository.URL, err)
+			continue
+		}
+
+		repo.Addons = adds
+		repositories.AddRepository(repo)
+	}
+
+	r.log.Info("- check duplicate ID addons alongside repositories")
+	repositories.ReviseBundleDuplicationInRepository()
+
+	r.log.Info("- check duplicates ID addons in existing AddonsConfiguration")
+	list, err := r.existingAddonsConfigurationList(addon)
+	if err != nil {
+		r.log.Errorf("cannot fetch AddonsConfiguration list: %s", err)
+		return exerr.Wrap(err, "while fetching addons configuration list")
+	}
+	repositories.ReviseBundleDuplicationInStorage(list)
+
+	if repositories.IsRepositoriesIDConflict() {
+		addon.Status.Phase = addonsv1alpha1.AddonsConfigurationFailed
+	} else {
+		addon.Status.Phase = addonsv1alpha1.AddonsConfigurationReady
+	}
+	r.log.Info("- AddonsConfiguration status: %s", addon.Status.Phase)
+
+	if addon.Status.Phase == addonsv1alpha1.AddonsConfigurationReady {
+		r.log.Info("- save ready bundles and charts in storage")
+		if err := r.saveBundle(internal.Namespace(addon.Namespace), repositories); err != nil {
+			return exerr.Wrap(err, "while saving ready bundles and charts in storage")
+		}
+	}
+
+	r.statusSnapshot(addon, repositories)
+	if _, err = r.updateAddonStatus(addon); err != nil {
+		r.log.Errorf("cannot update AddonsConfiguration %s: %v", addon.Name, err)
+		return exerr.Wrap(err, "while update AddonsConfiguration status")
+	}
+
+	if lastStatus.Phase == addonsv1alpha1.AddonsConfigurationReady && addon.Status.Phase == addonsv1alpha1.AddonsConfigurationReady {
+		deletedBundles, err := r.deleteOrphanBundles(addon.Namespace, addon.Status.Repositories, lastStatus.Repositories)
+		if err != nil {
+			return exerr.Wrap(err, "while deleting orphan bundles from storage")
+		}
+		for _, key := range deletedBundles {
+			if err := r.reprocessConflictingAddonsConfiguration(key, list); err != nil {
+				return exerr.Wrap(err, "while requesting processing of conflicting AddonsConfigurations")
+			}
+		}
+	}
+
+	r.log.Info("- ensure ServiceBroker")
+	if err := r.ensureBroker(addon); err != nil {
+		return exerr.Wrap(err, "while ensuring ServiceBroker")
+	}
+
+	return nil
+}
+
+func (r *ReconcileAddonsConfiguration) deleteAddonsProcess(addon *addonsv1alpha1.AddonsConfiguration) error {
+	r.log.Infof("Start delete AddonsConfiguration %s/%s process", addon.Name, addon.Namespace)
+
+	addonsCfgs, err := r.existingAddonsConfigurationList(addon)
+	if err != nil {
+		return exerr.Wrapf(err, "while listing AddonsConfigurations in namespace %s", addon.Namespace)
+	}
+
+	deleteBroker := true
+	for _, addon := range addonsCfgs.Items {
```

Comment:
If the deleted element is not in `ready` state then reprocess other addons is not necessary.